nemo/collections/multimodal/speech_llm/data/lhotse_dataset.py

```diff
@@ -0,0 +1,493 @@
+import copy
+import random
+from typing import Optional
+
+import numpy as np
+import torch.utils.data
+from lhotse.dataset.collation import collate_vectors as collate_vectors_lhotse
+from nemo.utils import logging
+
+
+def collate_vectors(items, max_length: int, padding_value):
+    vectors = collate_vectors_lhotse(items, padding_value=padding_value)
+    if max_length > vectors.size(1):
+        vectors = torch.cat(
+            [vectors, padding_value * torch.ones(vectors.size(0), max_length - vectors.size(1), dtype=vectors.dtype)],
```

Comment:
why do we need to enforce a static shape with padding for every example here?

Reply:
to be consistent with the behavior in mtron dataloader and AudioTextDataset in nemo/collections/multimodal/speech_llm/data/audio_text_dataset.py

Reply:
but that kills the purpose of doing bucketing, doesn't it? at least to some extent - as far as I can tell we only do this for transcripts, but keep audio shapes dynamic

Reply:
yes